From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Fri, 17 Oct 2014 17:23:52 +0200
Subject: Don't run main loop twice

Running the main loop twice segfaults on non i386 architectures even
with all libvirt and libvirt-glib code removed. So work around this by
manually iterating the loop to avoid the first quit.

Closes: #765566
---
 tests/test-events.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/tests/test-events.c b/tests/test-events.c
index ee7ff0b..6f65e76 100644
--- a/tests/test-events.c
+++ b/tests/test-events.c
@@ -91,11 +91,13 @@ static gboolean test_watch(gpointer user_data)
 
 static void test_remove_disabled_watch(void)
 {
+    GMainContext *context;
     main_loop = g_main_loop_new(NULL, FALSE);
+    context = g_main_loop_get_context(main_loop);
     watch_fd = open("/bin/true", O_RDONLY);
     g_idle_add(test_watch, NULL);
-    g_main_loop_run(main_loop);
-    g_main_loop_unref(main_loop);
+    g_main_context_iteration(context, TRUE);
+    g_main_context_iteration(context, TRUE);
     close(watch_fd);
     watch_fd = -1;
     g_assert_cmpint(watch_id, ==, -1);
